### ISSUE TITLE
x/vuln/cmd/govulncheck: cgo incompatibility

### ISSUE THREAD

**Issue by gbedoya** [golang/go#51819]
### What version of Go are you using (`go version`)?

<pre>
go version go1.18 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="[...]/.cache/go-build"
GOENV="[...]/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="[...]/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="[...]/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.18"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="[...]/podman/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build3833157872=/tmp/go-build -gno-record-gcc-switches"

</pre></details>

### What did you do?
Running `govulncheck` on packages that rely on cgo results in errors 

### What did you see?

```
➜  podman git:(main) govulncheck -json ./...   
[...]/podman/vendor/github.com//gpgme/data.go:7:8: could not import C (no metadata for C)
[...]/podman/vendor/github.com/proglottis/gpgme/data.go:150:13: invalid operation: cannot compare d.dh == nil (operator == not defined on untyped nil)
[...]/podman/vendor/github.com/proglottis/gpgme/gpgme.go:294:14: invalid operation: cannot compare c.ctx == nil (operator == not defined on untyped nil)
[...]/podman/vendor/github.com/proglottis/gpgme/gpgme.go:442:24: invalid operation: cannot compare key.k == nil (operator == not defined on untyped nil)
[...]/podman/vendor/github.com/containers/storage/drivers/btrfs/btrfs.go:15:8: could not import C (no metadata for C)
[...]/podman/vendor/github.com/containers/storage/pkg/devicemapper/devmapper_log.go:5:8: could not import C (no metadata for C)
dependencies/dependencies.go:7:2: import "github.com/onsi/ginkgo/ginkgo" is a program, not an importable package
govulncheck: packages contain errors
```
Details ...
https://github.com/containers/podman/blob/main/vendor/github.com/proglottis/gpgme/data.go#L7

How should `govulncheck` ignore `import "C"`?

**Comment by mknyszek** [comment]
CC @golang/security

**Comment by zpavlinovic** [comment]
This seems like a package loading issue. govulncheck uses the underlying build system to build the specified packages.

@jba

**Comment by zpavlinovic** [comment]
Note: the new version of govulncheck is now in [x/vuln/cmd/govulncheck](https://github.com/golang/vuln/tree/master/cmd/govulncheck). The previous version has been deleted.

**Comment by jba** [comment]
As a simple test, I created a Go program that uses cgo and ran govulncheck on it. It worked properly. This is using the standard `go` tool, with Go 1.18.

As a test to see if CGO_ENABLED was the culprit, I set `CGO_ENABLED=0`, but got a different error message. I wasn't able to generate the message "could not import C (no metadata for C)".

I think you'll have to give us steps to reproduce the bug. I can't make progress without it.
